bundles/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/internal/discovery/NikoHomeControlDiscoveryService.java

```diff
@@ -78,79 +80,121 @@ public void discoverDevices() {
+                    logger.debug("unrecognized action type {} for {} {}", nhcAction.getType(), deviceId, thingName);
             }
         });
+    }
 
+    private void discoverThermostatDevices(NikoHomeControlBridgeHandler bridgeHandler,
+            NikoHomeControlCommunication nhcComm) {
         Map<String, NhcThermostat> thermostats = nhcComm.getThermostats();
 
-        thermostats.forEach((thermostatId, nhcThermostat) -> {
+        thermostats.forEach((deviceId, nhcThermostat) -> {
             String thingName = nhcThermostat.getName();
             String thingLocation = nhcThermostat.getLocation();
-            addThermostatDevice(new ThingUID(THING_TYPE_THERMOSTAT, thingHandler.getThing().getUID(), thermostatId),
-                    thermostatId, thingName, thingLocation);
+            addDevice(new ThingUID(THING_TYPE_THERMOSTAT, bridgeHandler.getThing().getUID(), deviceId),
+                    CONFIG_THERMOSTAT_ID, deviceId, thingName, thingLocation);
         });
+    }
+
+    private void discoverMeterDevices(NikoHomeControlBridgeHandler bridgeHandler,
+            NikoHomeControlCommunication nhcComm) {
+        if (bridgeHandler instanceof NikoHomeControlBridgeHandler2) {
+            // disable discovery of NHC II energy meters to avoid overload in Niko Home Control cloud, can be removed
+            // when Niko solves their issue with the controller sending all live power data to their cloud
+            return;
+        }
+
+        Map<String, NhcMeter> meters = nhcComm.getMeters();
 
-        Map<String, NhcEnergyMeter> energyMeters = nhcComm.getEnergyMeters();
+        meters.forEach((deviceId, nhcMeter) -> {
+            String thingName = nhcMeter.getName();
+            String thingLocation = nhcMeter.getLocation();
 
-        energyMeters.forEach((energyMeterId, nhcEnergyMeter) -> {
-            String thingName = nhcEnergyMeter.getName();
-            String thingLocation = nhcEnergyMeter.getLocation();
-            addEnergyMeterDevice(new ThingUID(THING_TYPE_ENERGYMETER, thingHandler.getThing().getUID(), energyMeterId),
-                    energyMeterId, thingName, thingLocation);
+            switch (nhcMeter.getType()) {
+                case ENERGY_LIVE:
+                    addDevice(new ThingUID(THING_TYPE_ENERGYMETER_LIVE, bridgeHandler.getThing().getUID(), deviceId),
+                            METER_ID, deviceId, thingName, thingLocation);
+                    break;
+                case ENERGY:
+                    addDevice(new ThingUID(THING_TYPE_ENERGYMETER, bridgeHandler.getThing().getUID(), deviceId),
+                            METER_ID, deviceId, thingName, thingLocation);
+                    break;
+                case GAS:
+                    addDevice(new ThingUID(THING_TYPE_GASMETER, bridgeHandler.getThing().getUID(), deviceId), deviceId,
+                            METER_ID, thingName, thingLocation);
```

Comment:
I have the feeling you inverted the second and third parameter ?

Reply:
Indeed, thanks